velox/functions/lib/Re2Functions.cpp

```diff
@@ -978,7 +988,24 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
       // Ensures that pattern has a single contiguous stream of wildcard
       // characters.
       if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+        if (singleCharacterWildcardCount) {
+          return PatternMetadata{PatternKind::kGeneric, 0};
+        }
+        // Check if the pattern is of type kFixedCenter, so the use of regular
+        // expressions can be avoided to match the fixed string.
```

Comment:
It might be worth saying here that this is not the first time with % and _ and a fixed pattern has been found also. So if the rest of the characters are %, then we can use fixedCenter. Else if another string is found then kGeneric should be used.

Reply:
Removed this check and simplified it using a variable to track the count of wildcard character sequences, thanks for the feedback.